core/src/main/java/io/methvin/watcher/DirectoryWatcher.java

```diff
@@ -139,6 +139,15 @@ public void watch() {
             if (Files.isDirectory(childPath, NOFOLLOW_LINKS)) {
               if (!Boolean.TRUE.equals(fileTreeSupported)) {
                 registerAll(childPath);
+              } else {
+                Files.walkFileTree(childPath, new SimpleFileVisitor<Path>() {
+                  @Override
+                  public FileVisitResult visitFile(Path file, BasicFileAttributes attrs) throws IOException {
```

Comment:
Should this also visit directories? We are calling `onEvent` for the directory create so that seems to be the most consistent behavior.